### ISSUE TITLE
 Failure to connect to PandA device w/ 0.8 ophyd-async and 11.1 IOC

### ISSUE THREAD

**Issue by jwlodek** [bluesky/ophyd-async#635]
After updating all packages to the newest versions, the PandA device does not connect. I added a breakpoint near where the error is happening, and found the following:

```
Connecting to PandA #1
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(58)connect()
-> if set(entry) == {"d"}:
(Pdb) n
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(59)connect()
-> connector = self._filler.make_child_device(name)
(Pdb) n
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(60)connect()
-> connector.pvi_pv = entry["d"]
(Pdb) n
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(56)connect()
-> for name, entry in entries.items():
(Pdb) n
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(57)connect()
-> breakpoint()
(Pdb) n
> /home/xf31id/.ipython/profile_collection/venv/lib64/python3.11/site-packages/ophyd_async/epics/pvi/_pvi.py(58)connect()
-> if set(entry) == {"d"}:
(Pdb) entry
[None, {'d': 'XF:31ID1-ES{PANDA:1}:CALC1:PVI'}, {'d': 'XF:31ID1-ES{PANDA:1}:CALC2:PVI'}]
(Pdb) set(entry)
*** TypeError: unhashable type: 'dict'
(Pdb) pprint.pprint(entries)
{'bits': {'d': 'XF:31ID1-ES{PANDA:1}:BITS:PVI'},
 'calc': [None,
          {'d': 'XF:31ID1-ES{PANDA:1}:CALC1:PVI'},
          {'d': 'XF:31ID1-ES{PANDA:1}:CALC2:PVI'}],
 'clock': [None,
           {'d': 'XF:31ID1-ES{PANDA:1}:CLOCK1:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:CLOCK2:PVI'}],
 'counter': [None,
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER1:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER2:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER3:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER4:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER5:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER6:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER7:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:COUNTER8:PVI'}],
 'data': {'d': 'XF:31ID1-ES{PANDA:1}:DATA:PVI'},
 'div': [None,
         {'d': 'XF:31ID1-ES{PANDA:1}:DIV1:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:DIV2:PVI'}],
 'filter': [None,
            {'d': 'XF:31ID1-ES{PANDA:1}:FILTER1:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:FILTER2:PVI'}],
 'fmc_in': {'d': 'XF:31ID1-ES{PANDA:1}:FMC_IN:PVI'},
 'fmc_out': {'d': 'XF:31ID1-ES{PANDA:1}:FMC_OUT:PVI'},
 'inenc': [None,
           {'d': 'XF:31ID1-ES{PANDA:1}:INENC1:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:INENC2:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:INENC3:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:INENC4:PVI'}],
 'lut': [None,
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT1:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT2:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT3:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT4:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT5:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT6:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT7:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:LUT8:PVI'}],
 'lvdsin': [None,
            {'d': 'XF:31ID1-ES{PANDA:1}:LVDSIN1:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:LVDSIN2:PVI'}],
 'lvdsout': [None,
             {'d': 'XF:31ID1-ES{PANDA:1}:LVDSOUT1:PVI'},
             {'d': 'XF:31ID1-ES{PANDA:1}:LVDSOUT2:PVI'}],
 'outenc': [None,
            {'d': 'XF:31ID1-ES{PANDA:1}:OUTENC1:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:OUTENC2:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:OUTENC3:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:OUTENC4:PVI'}],
 'pcap': {'d': 'XF:31ID1-ES{PANDA:1}:PCAP:PVI'},
 'pcomp': [None,
           {'d': 'XF:31ID1-ES{PANDA:1}:PCOMP1:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:PCOMP2:PVI'}],
 'pgen': [None,
          {'d': 'XF:31ID1-ES{PANDA:1}:PGEN1:PVI'},
          {'d': 'XF:31ID1-ES{PANDA:1}:PGEN2:PVI'}],
 'pulse': [None,
           {'d': 'XF:31ID1-ES{PANDA:1}:PULSE1:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:PULSE2:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:PULSE3:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:PULSE4:PVI'}],
 'seq': [None,
         {'d': 'XF:31ID1-ES{PANDA:1}:SEQ1:PVI'},
         {'d': 'XF:31ID1-ES{PANDA:1}:SEQ2:PVI'}],
 'sfp3_evr': {'d': 'XF:31ID1-ES{PANDA:1}:SFP3_EVR:PVI'},
 'srgate': [None,
            {'d': 'XF:31ID1-ES{PANDA:1}:SRGATE1:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:SRGATE2:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:SRGATE3:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:SRGATE4:PVI'}],
 'system': {'d': 'XF:31ID1-ES{PANDA:1}:SYSTEM:PVI'},
 'ttlin': [None,
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN1:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN2:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN3:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN4:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN5:PVI'},
           {'d': 'XF:31ID1-ES{PANDA:1}:TTLIN6:PVI'}],
 'ttlout': [None,
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT1:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT2:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT3:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT4:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT5:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT6:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT7:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT8:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT9:PVI'},
            {'d': 'XF:31ID1-ES{PANDA:1}:TTLOUT10:PVI'}]}

```

Versions:

```
>>> import pandablocks_ioc                 
>>> pandablocks_ioc.__version__
'0.11.1'
```

```
(Pdb) import ophyd_async
(Pdb) ophyd_async.__version__
'0.8.0a3.dev1+gca56f74e.d20241101'
```

## Steps To Reproduce
1. Update ophyd async to most recent rc, PandA IOC to 11.1
2. Use device collector to instantiate a device.

## Acceptance Criteria
- Can connect to a PandA device.

**Comment by coretl** [comment]
Yes, we just spotted this ourselves, #625 (via #598) is needed to fix, as is PandABlocks-ioc 0.11.2

**Comment by jwlodek** [comment]
Can confirm that pulling the most recent versions now resolves this
